overlord/hookstate/hooks.go

```diff
@@ -98,6 +99,20 @@ func SetupPostRefreshComponentHook(st *state.State, snap, component string) *sta
 	return task
 }
 
+func SetupRemoveComponentHook(st *state.State, snap, component string) *state.Task {
+	hooksup := &HookSetup{
+		Snap:      snap,
+		Component: component,
+		Hook:      "remove",
+		Optional:  true,
```

Comment:
what's the IgnoreError setting on the snap remove hooks, we should be consistent?